apps/web/app/(app)/environments/[environmentId]/settings/profile/actions.ts

```diff
@@ -82,3 +82,17 @@ export async function updateAvatarAction(avatarUrl: string) {
 
   return await updateUser(session.user.id, { imageUrl: avatarUrl });
 }
+
+export async function removeAvatarAction() {
+  const session = await getServerSession(authOptions);
+
+  if (!session) {
+    throw new Error("Not authenticated");
+  }
+
+  if (!session.user.id) {
+    throw new Error("User not found");
+  }
+
```

Comment:
Can we here also delete the image from the storage?
For this we have a `deleteFile` method in `packages/lib/storage/service.ts`
But then we also need to check if the current user has environment access:
```
const isUserAuthorized = await hasUserEnvironmentAccess(session.user.id, environmentId);
```

Reply:
```
export async function removeAvatarAction() {
  const session = await getServerSession(authOptions);
  if (!session) throw new AuthorizationError("Not authorized");

  if (!session.user.id) {
    throw new Error("User not found");
  }

  const isUserAuthorized = await hasUserEnvironmentAccess(session.user.id, environmentId);
  if (!isUserAuthorized) throw new AuthorizationError("Not authorized");

  const { message, success, code } = await deleteFile(environmentId, accessType, fileName);

  if (success) {
    // revalidate cache
    storageCache.revalidate({ fileKey: `${environmentId}/${accessType}/${fileName}` });
    return responses.successResponse(message);
  }

  if (code === 404) {
    return responses.notFoundResponse("File", "File not found");
  }

  return await updateUser(session.user.id, { imageUrl: null });
}
```

So I think it should look something like this, but do you know how to get the filename? Is it just the last part of the url?